src/main/java/org/mule/extensions/vm/internal/operations/VMOperations.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extensions.vm.internal.operations;
+
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+import static org.mule.extensions.vm.api.VMError.EMPTY_QUEUE;
+import static org.mule.extensions.vm.api.VMError.QUEUE_TIMEOUT;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import org.mule.extensions.vm.api.VMMessageAttributes;
+import org.mule.extensions.vm.internal.QueueDescriptor;
+import org.mule.extensions.vm.internal.ReplyToCommand;
+import org.mule.extensions.vm.internal.VMConnectorQueueManager;
+import org.mule.extensions.vm.internal.connection.VMConnection;
+import org.mule.runtime.api.component.location.ComponentLocation;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.lifecycle.Startable;
+import org.mule.runtime.api.lifecycle.Stoppable;
+import org.mule.runtime.api.metadata.TypedValue;
+import org.mule.runtime.api.scheduler.Scheduler;
+import org.mule.runtime.core.api.scheduler.SchedulerService;
+import org.mule.runtime.core.api.util.queue.Queue;
+import org.mule.runtime.extension.api.annotation.error.Throws;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Content;
+import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
+import org.mule.runtime.extension.api.exception.ModuleException;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+public class VMOperations implements Startable, Stoppable {
```

Comment:
Sadly not. They were all NB but then I realised that it has to support XA and the fun stopped.